tests/numerics/coupling_matrix_test.C

```diff
@@ -169,7 +169,7 @@ private:
         for (unsigned int pj = 0; pj != sizeof(jvals)/isize; ++pj)
           {
             CPPUNIT_ASSERT(ccr_it != ccr.end());
-            CPPUNIT_ASSERT_EQUAL(*ccr_it, jvals[pj]);
+            CPPUNIT_ASSERT_EQUAL((unsigned int) *ccr_it, jvals[pj]);
```

Comment:
Tangentially related minor point: I don't know why but we seem to have a bunch of C-style casts in the unit tests when a `static_cast` is the intent. So that's an Effective C++ thing that I will make a separate PR for.